src/main/java/de/tum/in/www1/artemis/web/rest/ComplaintResource.java

```diff
@@ -148,30 +118,16 @@ public ResponseEntity<Complaint> createComplaintForExamExercise(@PathVariable Lo
                 .headers(HeaderUtil.createEntityCreationAlert(applicationName, true, entityName, savedComplaint.getId().toString())).body(savedComplaint);
     }
 
-    private void validateNewComplaint(Complaint complaint) {
-        if (complaint.getId() != null) {
-            throw new BadRequestAlertException("A new complaint cannot already have an id", COMPLAINT_ENTITY_NAME, "idExists");
-        }
-
-        if (complaint.getResult() == null || complaint.getResult().getId() == null) {
-            throw new BadRequestAlertException("A complaint can be only associated to a result", COMPLAINT_ENTITY_NAME, "noresultid");
-        }
-
-        if (complaintRepository.findByResultId(complaint.getResult().getId()).isPresent()) {
-            throw new BadRequestAlertException("A complaint for this result already exists", COMPLAINT_ENTITY_NAME, "complaintexists");
-        }
-    }
-
     /**
-     * Get complaints/submissions/{submissionId} get a complaint associated with a result of the submission "id"
+     * GET complaints: get a complaint associated with a result of the submission "id"
      *
      * @param submissionId the id of the submission for whose results we want to find a linked complaint
      * @return the ResponseEntity with status 200 (OK) and either with the complaint as body or an empty body, if no complaint was found for the result
      */
-    @GetMapping("complaints/submissions/{submissionId}")
+
+    @GetMapping(value = "complaints", params = { "submissionId" })
```

Comment:
Is this annotation usage of the `@GetMapping` with the params parameter intended? 🤔
Because we never used it like this in our code base.

Reply:
Yes, it is intended. It is used for distinguish between the endpoints that has the same URL but different query parameters.